add_codeowners.py

```diff
@@ -0,0 +1,498 @@
+
+from ghapi.all import GhApi
+
+from helpers import DryRunner
+
+
+class CodeownerGroup(str, Enum):
+    ADMIN = "software-admin"
+    PYTHON = "python-reviewers"
+    TWINCAT = "twincat-reviewers"
+    EPICS = "epics-reviewers"
+    SHELL = "shell-reviewers"
+    CPP = "cpp-reviewers"
+    VACUUM_SME = "vacuum-sme"
+    LASER_SME = "laser-sme"
+    MOTION_SME = "motion-sme"
+    PMPS_SME = "pmps-sme"
+    UI_SME = "ui-sme"
+    LCLS_NAMING = "lcls-naming-council"
+    TMO = "tmo-members"
+    CXI = "cxi-members"
+    LAS = "las-members"
+    MEC = "mec-members"
+    MFX = "mfx-members"
+    RIX = "rix-members"
+    TXI = "txi-members"
+    XCS = "xcs-members"
+    XPP = "xpp-members"
+
+
+@dataclass
+class RepoOwnerSettings:
+    owner: str
+    repo_name: str
+    sme_owners: List[CodeownerGroup]
+    lang_owners: List[CodeownerGroup]
+    area_owners: List[CodeownerGroup]
+
+    @classmethod
+    def from_dict(cls, source: dict):
+        settings = cls(
+            owner=source["owner"],
+            repo_name=source["repo_name"],
+            sme_owners=[CodeownerGroup(value.strip()) for value in
+                        source["sme_owners"].split(',') if value],
+            area_owners=[CodeownerGroup(value.strip()) for value in
+                         source["area_owners"].split(',') if value],
+            lang_owners=[CodeownerGroup(value.strip()) for value in
+                         source["lang_owners"].split(',') if value],
+        )
+
+        settings.sme_owners.sort()
+        settings.area_owners.sort()
+        settings.lang_owners.sort()
+
+        return settings
+
+
+GROUP_TO_EXT: Dict[CodeownerGroup, str] = {
+    CodeownerGroup.PYTHON: '*.py*',
+    CodeownerGroup.TWINCAT: '*.{plcproj,sln,TcDUT,TcGVL,TcPOU,TcTTO,tmc,tsproj,xti}',
+    CodeownerGroup.CPP: '*.cpp',
+    CodeownerGroup.SHELL: '*.sh',
+    CodeownerGroup.EPICS: '*.{archive,autosave,cmd,db,dbd,edl,ioc,proto,req,'
+                          'sub-arch,sub-req,substitutions,tpl-arch,tpl-req}',
+}
+
+
+def create_codeowners_file(settings: RepoOwnerSettings) -> str:
+    """
+    Create codeowners file based on RepoOwnerSettings
+
+    See https://confluence.slac.stanford.edu/x/u5Q9Hw for details.
+    By default:
+    - "default" = sme_owners + area_owners
+    - .github owned by admins only
+    - Default (*): default
+    - language specific files: default + lang_owners
+
+    Returns content encoded as base64 utf-8, in order to ensure the
+    payload is appropriate for github's REST API
+    """
+    lines = []
+
+    # gather groups
+    default_owners = list(set(settings.sme_owners + settings.area_owners))
+    if not default_owners:
+        default_owners = [CodeownerGroup.ADMIN]
+    default_groups = ["@pcdshub/" + grp for grp in default_owners]
+
+    # .github admin group
+    lines.append("# github folder holds administrative files")
+    lines.append(f".github/** @pcdshub/{CodeownerGroup.ADMIN}")
+    lines.append("")
```

Comment:
Change request:
Putting any group prior to the "default" group means that this group will never be called upon for review. I think admin in particular should be the last group to make sure we always vet changes to the .github stuff.

Reply:
I've moved the .github section to the end so it always takes precedence.  This will, however, prevent area/language/sme groups from being summoned

Reply:
but to that we'll respond: split up your PRs